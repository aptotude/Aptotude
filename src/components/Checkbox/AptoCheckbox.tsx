import './aptoCheckbox.scss';
export { Checkbox as AptoCheckbox } from 'semantic-ui-react';
