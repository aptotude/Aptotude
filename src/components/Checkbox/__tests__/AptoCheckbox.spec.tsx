import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoCheckbox } from '../AptoCheckbox';

afterEach(() => {
  cleanup();
});

describe('AptoCheckbox', () => {
  it('Renders Component', () => {
    const { container } = render(<AptoCheckbox />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders with name/value', () => {
    const { container } = render(<AptoCheckbox name="foo" value="bar" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders checked', () => {
    const { container } = render(
      <AptoCheckbox checked={true} name="foo" value="bar" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders disabled', () => {
    const { container } = render(
      <AptoCheckbox disabled={true} name="foo" value="bar" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
