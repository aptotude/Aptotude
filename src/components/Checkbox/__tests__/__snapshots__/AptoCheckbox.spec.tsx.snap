// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoCheckbox Renders Component 1`] = `
<div
  class="ui fitted checkbox"
>
  <input
    class="hidden"
    readonly=""
    tabindex="0"
    type="checkbox"
    value=""
  />
  <label />
</div>
`;

exports[`AptoCheckbox Renders checked 1`] = `
<div
  class="ui checked fitted checkbox"
>
  <input
    checked=""
    class="hidden"
    name="foo"
    readonly=""
    tabindex="0"
    type="checkbox"
    value="bar"
  />
  <label />
</div>
`;

exports[`AptoCheckbox Renders disabled 1`] = `
<div
  class="ui disabled fitted checkbox"
>
  <input
    class="hidden"
    disabled=""
    name="foo"
    readonly=""
    tabindex="-1"
    type="checkbox"
    value="bar"
  />
  <label />
</div>
`;

exports[`AptoCheckbox Renders with name/value 1`] = `
<div
  class="ui fitted checkbox"
>
  <input
    class="hidden"
    name="foo"
    readonly=""
    tabindex="0"
    type="checkbox"
    value="bar"
  />
  <label />
</div>
`;
